lib/std/setutils.nim

```diff
@@ -93,6 +134,19 @@ proc `-+-`*[T](x, y: set[T]): set[T] {.inline.} =
     assert {1, 2, 3} -+- {2, 3, 4} == {1, 4}
   result = symmetricDifference(x, y)
 
+proc toggle*[T](x: var set[T], y: T) {.inline.} =
+  ## Toggles the element `y` in the set `x`.
+  ## If the element `y` is in `x`, it is excluded from `x`;
+  ## otherwise it is included.
+  ## Equivalent to `x = symmetricDifference(x, {y})`.
+  runnableExamples:
+    var x = {1, 2, 3}
+    x.toggle(2)
+    x.toggle(3)
+    x.toggle(4)
+    assert x == {1, 4}
+  x = symmetricDifference(x, {y})
```

Comment:
For large sets this is worse than checking if the value is in a set then including/excluding.

Reply:
you are right, so i will just do
```nim
if y in x: x.excl(y)
else: x.incl(y)
```
because in codegen when the set has odd sizes it may be an array of U8 even if the size is less or equal than 8 bytes